app/utils/theGraph.js

```diff
@@ -200,11 +200,16 @@ export const postsForSearch = async (text, single) => {
       text: query,
     },
   });
-  return posts?.data?.postSearch.filter(
-    post =>
-      !post.isDeleted &&
-      (single ? Number(post.communityId) === Number(single) : true),
-  );
+  return posts?.data?.postSearch
+    .slice(0, 100)
+    .sort((a, b) => {
+      return b.postTime - a.postTime;
+    })
```

Comment:
you need to transfer the sorting and the limit to the request so that it comes from the back in the desired format